osu.Game.Rulesets.Osu/Skinning/Legacy/LegacySpinnerApproachCircle.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Skinning.Legacy
+{
+    public class LegacySpinnerApproachCircle : CompositeDrawable
+    {
+        private DrawableSpinner drawableSpinner;
+
+        [CanBeNull]
+        private Sprite approachCircle;
+
+        [BackgroundDependencyLoader]
+        private void load(DrawableHitObject drawableHitObject, ISkinSource source)
+        {
+            drawableSpinner = (DrawableSpinner)drawableHitObject;
+
+            AutoSizeAxes = Axes.Both;
+
+            var spinnerProvider = source.FindProvider(s =>
+                s.GetTexture("spinner-circle") != null ||
+                s.GetTexture("spinner-top") != null);
+
+            if (spinnerProvider is DefaultLegacySkin)
+                return;
+
+            InternalChild = approachCircle = new Sprite
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                Texture = source.GetTexture("spinner-approachcircle"),
+                Scale = new Vector2(1.86f),
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            drawableSpinner.ApplyCustomUpdateState += updateStateTransforms;
+            updateStateTransforms(drawableSpinner, drawableSpinner.State.Value);
+        }
+
+        private void updateStateTransforms(DrawableHitObject drawableHitObject, ArmedState state)
+        {
+            if (!(drawableHitObject is DrawableSpinner spinner))
+                return;
```

Comment:
This really ought to be a straight cast. If this is ever hit then that sounds like catastrophic failure.

Reply:
`drawableHitObject` could be nested objects of the spinner, like `DrawableSpinnerTick` and otherwise. Maybe a switch case would make sense?

Reply:
Ok, I keep forgetting this also gets called on nested. Would leave a comment at most.

Reply:
I've changed it to a switch case, I think that would indicate it more nicely.